src/StackedTable.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import propTypes from 'prop-types'
+import { instanceOfComponent } from '@dhis2/prop-types'
+
+import { extractHeaderLabels } from './StackedTable/extractHeaderLabels.js'
+import { Provider } from './StackedTable/TableContext.js'
+import { StackedTableBody } from './StackedTableBody.js'
+import { StackedTableFoot } from './StackedTableFoot.js'
+import { StackedTableHead } from './StackedTableHead.js'
+import { colors } from './theme.js'
+
+const childPropType = propTypes.oneOfType([
+    instanceOfComponent(StackedTableHead),
+    instanceOfComponent(StackedTableBody),
+    instanceOfComponent(StackedTableFoot),
+])
+
+const childrenPropType = propTypes.oneOfType([
+    childPropType,
+    propTypes.arrayOf(childPropType),
+])
+
+export const Table = ({ children, className, dataTest }) => (
+    <table className={className} data-test={dataTest}>
+        {children}
+
+        <style jsx>{`
+            table {
+                display: block;
+                border: 0;
+                background-color: #ffffff;
+                min-width: 100%;
+                text-align: left;
+                border-collapse: collapse;
+                vertical-align: top;
+                color: ${colors.grey900};
+            }
+        `}</style>
+    </table>
+)
+
+Table.propTypes = {
+    children: childrenPropType.isRequired,
+    className: propTypes.string,
+    dataTest: propTypes.string,
+}
+
+export const StackedTable = ({ children, className, dataTest }) => {
+    const headerLabels = extractHeaderLabels(children)
+
+    return (
+        <Provider value={{ headerLabels }}>
```

Comment:
We should pass the `headerLabels` as props to the children as well so that it is possible to use custom child components without relying on the provider (which is not exposed publicly).

Reply:
I've moved the context up to the row, which now supplies the cells with the `headerLabels`, which allows custom table cells (I've added a story).